### PR TITLE
Rollup (3603, 3646 and 3672)

### PR DESCRIPTION
**Pull request by nbougalis**
This PR incorporates and supersedes three separate PRs. If merged, it will close #3603, close #3646 and close #3672.

### CODE REVIEW

**Review by seelabs — APPROVED** [review]
:+1: My approval from #3646 can carry over to this PR.

File: .travis.yml [hunk 1]
```diff
@@ -323,7 +311,7 @@ matrix:
         - export TEST_EXTRA_ARGS="--unittest-ipv6"
     - <<: *macos
       osx_image: xcode11.2
-      name: xcode11, debug
+      name: xcode11.2, debug
```
Comment (cjcobb23) [inline .travis.yml:314]: There are two builds named "xcode11.2, debug" now (here and on line 266), and I think they are identical. We only need one of these right?

**Review by HowardHinnant — APPROVED** [review]

**Review by gregtatcam — COMMENTED** [review]
#3672 looks fine. I still have #3646 on my plate though.

File: .travis.yml (hunk 1 above)
Comment (nbougalis) [reply .travis.yml:314]: I believe on runs IPv6-specific unit tests. I don't know that really gets us much and we could remove it.

**Review by gregtatcam — APPROVED** [review]
:+1: for #3646, #3672

**Review by scottschurr — COMMENTED** [review]
https://github.com/ripple/rippled/pull/3603 still looks good with the exception of concerns about the API change.  Here's where the API change was first noted in 3606: https://github.com/ripple/rippled/pull/3603#discussion_r487348481

Here's the most recent comment regarding the API change, which includes an (untested) patch that puts the API change under API versioning: https://github.com/ripple/rippled/pull/3603#issuecomment-733140844

I'm okay if:
- We decide that the API change is non-breaking, or
- Decide the API change _is_ breaking but that breakage is justified and we'll just document it, or
- We'll modify this pull request to back out the API change, or
- We'll modify this pull request to version the API change, or
- We'll have a follow-on pull request to version the API change before the next release candidate.

But I'm uncomfortable with just moving ahead and not stating explicitly how we intend to address the API change.  Please, let's have an explicit call on how the API change will be handled.

Thanks.

File: src/ripple/rpc/handlers/Peers.cpp [hunk 2]
```diff
@@ -32,17 +32,31 @@ doPeers(RPC::JsonContext& context)
...
 
+    jvResult[jss::peers] = context.app.overlay().json();
+
+    // Legacy support
+    if (context.apiVersion == 1)
     {
-        jvResult[jss::peers] = context.app.overlay().json();
+        for (auto& p : jvResult[jss::peers])
+        {
+            if (auto s = p[jss::track].asString(); !s.empty())
```
Comment (scottschurr) [inline src/ripple/rpc/handlers/Peers.cpp:42]: This code has a subtle bug courtesy of our JSON library.  By _examining_ `jss::track` in this way you have injected `jss::track` into object `s` with a value of `null`.  You can avoid the bug with brute force by doing something like this:
```
        for (auto& p : jvResult[jss::peers])
        {
            if (p.isMember(jss::track)) // Important test
            {
                if (auto s = p[jss::track].asString(); !s.empty())
                {
                    if (s == "diverged")
                        p["sanity"] = "insane";
                    if (s == "unknown")
                        p["sanity"] = "unknown";
                }
            }
        }
```
Or perhaps you have a more clever suggestion.

**Review by scottschurr — CHANGES_REQUESTED** [review]
I think there's a bug in the API patch.  Take a look and see if you agree.  I think this is really close...

File: src/ripple/rpc/handlers/Peers.cpp (hunk 2 above)
Comment (nbougalis) [reply src/ripple/rpc/handlers/Peers.cpp:42]: Ooof, you're right. I totally missed that, thanks. I'll adopt your approach.

Comment (nbougalis) [reply src/ripple/rpc/handlers/Peers.cpp]: I've fixed this thusly: 

https://github.com/ripple/rippled/blob/c52a3db8db49214385ab9f3f1d2944a2412e4222/src/ripple/rpc/handlers/Peers.cpp#L38-L52

**Review by scottschurr — APPROVED** [review]
:+1:

**Pull request closed without merging**